File: 백준/Bronze/14568. 2017 연세대학교 프로그래밍 경시대회/2017 연세대학교 프로그래밍 경시대회.py
N = int(input())

# n = y + k
# t % 2 = 0

# N - t = y + y + k
# 2y : tmp - k

answer = set()
for a in range(2, N-1, 2):
    tmp = N - a
    for k in range(2, tmp):
        if (tmp - k) % 2 == 0:
            b = (tmp - k) // 2
            if b > 0:
                c = b + k
                answer.add((a, b, c))

print(len(answer))

File: 백준/Bronze/14568. 2017 연세대학교 프로그래밍 경시대회/2017 연세대학교 프로그래밍 경시대회.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Set;
import java.util.List;
import java.util.HashSet;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        Set<List<Integer>> answer = new HashSet<>();
        for (int a = 2; a < N-1; a += 2) {
            int tmp = N - a;
            for (int k = 2; k < tmp; k++) {
                if ((tmp - k) % 2 == 0) {
                    int b = (tmp - k) / 2;
                    if (b > 0) {
                        int c = b + k;
                        answer.add(Arrays.asList(a, b, c));
                    }
                }
            }
        }
        System.out.print(answer.size());
    }
}
